### ISSUE TITLE
JSX Children Tuple with single element

### ISSUE THREAD

**Issue by plusgut** [Microsoft/TypeScript#30918]
**Search Terms:**
When A single element is given as a Child, it is not handled as an Array.

**Code**

```ts
const Component = (props: { children: [ JSX.Element ] }) => props.children;

const test = <Component><element /></Component>; // Should not throw an error
```

**Expected behavior:**
There should not be a completely different behaviour from TypeScript when a single Child is present in a Tuple.

**Actual behavior:**
The following error is thrown: _This JSX tag's 'children' prop expects type '[Element]' which requires multiple children, but only a single child was provided_

I am aware that React behaves that way, but the JSX-Language states that everything between _JSXOpeningElement_ and _JSXSelfClosingElement_ are _JSXChildren_. https://github.com/facebook/jsx

**Comment by RyanCavanaugh** [comment]
`{ children: [ JSX.Element ] }` is basically an impossible contract to satisfy.

I don't understand what you expect to happen here. Your function says that this code is legal:
```ts
const Component = (props: { children: [ JSX.Element ] }) => props.children.slice(0);

const test = <Component><element /></Component>; // Should not throw an error
```
But that code is going to throw an error at runtime. Why should something that manifestly defies the specified types *pass* the type checking?

**Comment by weswigham** [comment]
FYI: Impossible in `react`, but fully possible (probable, even) in other JSX frameworks. This is a place where we have some unfortunate `react`-unique logic baked in, since, AFAIK, at the time we didn't even have tuple types but still wanted to allow validation against singular children?

**Comment by plusgut** [comment]
@RyanCavanaugh 
It would throw a runtime error in react, but not necessarily in other frameworks.
And especially when we look at the jsx specifications itself, React is doing it wrong.

@weswigham Exactly! Don't you agree that there should be the option, to opt-out of these react-specific things?
